src/services/services.ts

```diff
@@ -2912,6 +2919,63 @@ module ts {
 
                 return filteredMembers;
             }
+
+            function isExternalModuleName(token: Node): boolean {
+                if (token && token.parent && token.kind === SyntaxKind.StringLiteral && isInStringOrRegularExpressionOrTemplateLiteral(token)) {
+                    return token.parent.kind === SyntaxKind.ImportDeclaration ||
+                        token.parent.kind === SyntaxKind.ExportDeclaration ||
+                        token.parent.kind === SyntaxKind.ExternalModuleReference;
+                }
+                return false;
+            }
+
+            function getExternalModulecompletions(location: Node): CompletionInfo {
+                var entries: CompletionEntry[] = [];
+                var currentSourceFile = location.getSourceFile();
+                var currentDirectoryPath = getDirectoryPath(normalizePath(currentSourceFile.fileName));
+                var isInAmbientContext = ts.isInAmbientContext(location);
+
+                forEach(program.getSourceFiles(), sourceFile => {
+                    if (isExternalModule(sourceFile)) {
+                        // Only consider these if current request is not in an ambient context,
+                        // and in a diffrent file
+                        if (!isInAmbientContext && sourceFile !== currentSourceFile) {

```

Comment:
yeah.. but chances are you have all your dependencies in ambients, and you will reference them but not the opposite. it just to limit clutter.